crawler/rabbit/async_consumer.py

```diff
@@ -25,16 +25,14 @@ class AsyncConsumer(object):
     commands that were issued and that should surface in the output as well.
     """
 
-    def __init__(self, server_details, queue):
+    def __init__(self, server_details, queue, process_message):
         """Create a new instance of the consumer class, passing in the AMQP
         URL used to connect to RabbitMQ.
-        :param bool use_ssl: Whether to use SSL when connecting to the AMQP endpoint.
-        :param str host: The AMQP host to connect to.
-        :param int port: The AMQP port to connect with.
-        :param str username: The AMQP username with read access to the queue.
-        :param str password: The AMQP password for the username.
-        :param str password: The AMQP virtual host to consume from.
+        :param RabbitServerDetails server_details: The RabbitMQ server connection details.
         :param str queue: The AMQP queue to consume from.
+        :param func process_message: A function to call with details of any messages consumed from the queue.
+                                     This function will be passed the message headers, message body and a function
+                                     to call to acknowledge that the message was processed.
```

Comment:
It could be an instance of a class that handles the process, would it simplify also the arguments?

Reply:
Actually this comment is out of date.  I use the return value after calling the function so I'll update the comment.